libs/src/utils/network.js

```diff
@@ -30,39 +30,47 @@ async function timeRequest (request, timeout = null) {
 }
 
 /**
- * Fetches multiple urls and times each request and returns the results sorted by
- * lowest-latency.
+ * Fetches multiple urls and times each request and returns the results
+ * filtered for healthy and sorted by lowest-latency if requested
  * @param {Array<Object>} requests [{id, url}, {id, url}]
+ * @param {string} currentVersion
  * @param {number?} timeout ms applied to each individual request
  * @returns { Array<{url, response, millis}> }
  */
-async function timeRequests (requests, timeout = null) {
+async function timeRequests (requests, currentVersion, timeout = null, equivalencyDelta = null) {
   let timings = await Promise.all(requests.map(async request =>
     timeRequest(request, timeout)
   ))
 
+  timings = timings.filter(timing => timing.response !== null)
+
+  timings = timings.sort((a, b) => {
+    // Sort by version only if behind current on-chain version
+    const aVersion = a.response.data.data.version
```

Comment:
trying to think through whether we can combine this method with the one below, or at least give them more descriptive function names, because they're similar, but slightly different.

We also have some tests for these fns, can we add some more? https://github.com/AudiusProject/audius-protocol/blob/master/libs/src/utils/network.test.js to make the difference more clear maybe?

Reply:
good call, will def make names more clear
i don't think `timeRequests` is used anywhere, so honestly would be simplest for me to just consolidate this into one
originally i didnt do that for back-compat but 1) i already changed it and 2) doesnt seem to be used anywhere

also nice, didnt see we have unit tests for this file as well, will add 👍

Reply:
yeah agreed with ray. this logic looks good. if we could encapsulate all the semver logic in this file into one function that would be great. adding more branches to this already complex flow is probably the wrong direction here